### ISSUE TITLE
[Feature request] Music (possibly midi) export

### ISSUE THREAD

**Issue by mward25** [kmatheussen/radium#1404]
I'd really like some way to get music out of radium and into sheet music. This could be via musicxml, ly (lilypond), midi, or any format that can be converted to musical data that I can import into musescore without to much work.

I figure the general implementation from the user's perspective would go something like this:
UI Implementation:
 - A button under "export soundfile(s)..." labeled "export [format]".
 - A file manager menu pops up asking where you wan't the file to be exported
 - When you click the Save button, the song exports

Keybinding:
 - I figure this does not need a key binding

Things that in general I'd like to have with this feature:
 - Note export
 - Time signature export
 
Things that I in general are indifferent about with this feature:
 - Tempo tracking
 - Exporting multiple seqtracks as one file
 
Anyway, have a great rest of your day.

**Comment by kmatheussen** [comment]
Yes, I think export MIDI would be more natural to implement before any type of score format. In the old days, when Radium was only about MIDI, you could create scores from .rad files using Common Music Notation, but it might be less work implementing MIDI export than reviving that code.

**Comment by mward25** [comment]
I figured out how to do this utilizing Radium's built in Pure Data. 

Here's the patch:
[export_midi.pd.gz](https://github.com/kmatheussen/radium/files/12187529/export_midi.pd.gz)
And here's my little test radium file:
[midi_export_test_embeded_samples.rad.gz](https://github.com/kmatheussen/radium/files/12187540/midi_export_test_embeded_samples.rad.gz)
And finally, here is the exported midi from that file:
[exported_midi.midi.gz](https://github.com/kmatheussen/radium/files/12187581/exported_midi.midi.gz)

NOTE: I have not yet figured out how to have a Pure Data instrument open a specific patch by default, so in this example you will have to do so manually.

I'm satisfied with the results. Should I close this issue?

**Comment by kmatheussen** [comment]
Is this better than sending midi out to another sequencer, which records
the midi messages?

**Comment by kmatheussen** [comment]
You can also use jack transport to make sure both radium and the
sequencer-program starts playing and recording at the same time.
Maybe qtractor, ardour, muse, rosegarden, bitwig, etc. could be used for
that.

**Comment by mward25** [comment]
I'd tried Qtractor, Ardour, and Muse, and had various problems with each, (normally involved with them finding the correct midi ports, or syncing properly with jack). 

However I hadn't tried Rosegarden before, so I gave that a try and it worked great; using Rosegarden in conjunction with Radium is exactly what I wanted/needed. Thanks for the recommendation. 

I see no advantage to using puredata, using Rosegarden works much better, and makes more since for what I want (because it has built in scoring).

**Comment by kmatheussen** [comment]
Great to hear! I see one advantage with the pd-solution though, and that's
timing. I guess it's possible to use jack-midi somehow (Radium supported it
directly before, but not anymore, so now you have to use a third party
plugin for sending midi to jack), and then you will get perfect timing, but
currently Alsa midi is used, and then timing is not guaranteed to be
perfect. If it makes any notable difference, I don't know though.

**Comment by kmatheussen** [comment]
MIDI export has been implemented and will be included in upcoming Radium V8.
